src/output/plugins/OSXOutputPlugin.cxx

```
@@ -103,6 +107,37 @@ OSXOutput::OSXOutput(const ConfigBlock &block)
 	sync_sample_rate = block.GetBlockValue("sync_sample_rate", false);
 }
 
+int
+OSXOutput::GetVolume()
+{
+	AudioUnitParameterValue dvolume;
+	char errormsg[1024];
+
+	OSStatus status = AudioUnitGetParameter(au, kHALOutputParam_Volume,
+			kAudioUnitScope_Global, 0, &dvolume);
+	if (status != noErr) {
+		osx_os_status_to_cstring(status, errormsg, sizeof(errormsg));
+		FormatWarning(osx_output_domain, "unable to get volume: %s", errormsg);
```

Throw an exception instead of logging a warning.